src/MongoDB.Driver.Core/Core/Configuration/ConnectionString.cs

```diff
@@ -110,14 +110,19 @@ public sealed class ConnectionString
         /// Initializes a new instance of the <see cref="ConnectionString" /> class.
         /// </summary>
         /// <param name="connectionString">The connection string.</param>
-        public ConnectionString(string connectionString)
+        public ConnectionString(string connectionString) : this(connectionString, DnsClientWrapper.Instance)
```

Comment:
it will allow to add more flexible tests for resolve logic

Reply:
Ack